src/EditorFeatures/CSharpTest/Completion/CompletionProviders/XmlDocumentationCommentCompletionProviderTests.cs

```diff
@@ -1084,6 +1134,55 @@ static void Main(string[] args)
 ", "args");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task ParamNames()
+        {
+            await VerifyItemExistsAsync(@"
+/// <param name=""$$""/>
+static void Foo(string str)
+{
+}
+", "str");
+        }
+
+        [WorkItem(17872, "https://github.com/dotnet/roslyn/issues/17872")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TypeParamRefNames()
+        {
+            var text = @"
+public class Outer<TOuter>
+{
+    public class Inner<TInner>
+    {
+        /// <summary>
+        /// <typeparamref name=""$$""/>
+        /// </summary>
+        public int Method<TMethod>(T green) { }
+    }
+}";
+
+            await VerifyItemsExistAsync(text, "TOuter", "TInner", "TMethod");
```

Comment:
Do these tests validate that all those items still show up, even if they're partially typed?  i.e. if ```"T"``` is already there, and someone tries to bring up completion, that should still work.